astropy/utils/data.py

```diff
@@ -1064,6 +1064,32 @@ def download_file(remote_url, cache=False, show_progress=True, timeout=None):
 
     return local_path
 
+def is_url_in_cache(url_key):
+    """
+    Check if a download from ``url_key`` is in the cache.
+
+    Parameters
+    ----------
+    url_key : string
+        The URL retrieved
+
+    Returns
+    -------
+    in_cache : bool
+        `True` if a download from ``url_key`` is in the cache
+    """
+    # The code below is modified from astropy.utils.data.download_file()
+    try:
+        dldir, urlmapfn = _get_download_cache_locs()
+    except (IOError, OSError) as e:
+        msg = 'Remote data cache could not be accessed due to '
+        estr = '' if len(e.args) < 1 else (': ' + str(e))
+        warn(CacheMissingWarning(msg + e.__class__.__name__ + estr))
+        return False
+    with _open_shelve(urlmapfn, True) as url2hash:
+        if url_key in url2hash:
+            return True
+    return False
```

Comment:
Although, gosh, now that I look at it this exact same code is duplicated in a number of places.

Seems to me this could be reworked into a simple (maybe internal only, for now) utility function to just get the `url2hash` map (or return an empty-dict along with the above warning if not found).

Then your function could just be:
```python
def is_url_in_cache(url_key):
    return url_key in _get_cache_url_map()
```
or something like that.

Up to you though, we can do that later if not in this PR.